idx/idx.go

```diff
@@ -112,4 +173,7 @@ type MetricIndex interface {
 	Find(int, string, int64) ([]Node, error)
 	List(int) []Archive
 	Prune(int, time.Time) ([]Archive, error)
+	TagList(int) []string
+	Tag(int, string, int64) map[string]uint32
+	IdsByTagExpressions(int, []string, int64) (map[MetricID]struct{}, error)
```

Comment:
`IdsByTagExpressions` follows a different naming pattern. none of the others (e.g. `Find`, `List`) include the return type in the method name, or the fact that they accept patterns/queries/expressions as input shouldn't go into the method name either (e.g. see `Find` also).  so this can just be `FindByTag`

Reply:
done https://github.com/grafana/metrictank/commit/0adb706cb059bd8b0422bf38ef76c59cc6c61274